File: scripts/test_solutions/1046_last_stone_weight.js
var lastStoneWeight = function(stones) {
    while (stones.length > 1) {
        stones.sort((a, b) => b - a);
        const diff = stones[0] - stones[1];
        stones = stones.slice(2);
        if (diff > 0) stones.push(diff);
    }
    return stones.length ? stones[0] : 0;
};

File: scripts/test_solutions/1046_last_stone_weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        priority_queue<int> heap(stones.begin(), stones.end());
        while (heap.size() > 1) {
            int a = heap.top(); heap.pop();
            int b = heap.top(); heap.pop();
            if (a != b) {
                heap.push(a - b);
            }
        }
        return heap.empty() ? 0 : heap.top();
    }
};
